src/Components/DonationFormModal.js

```diff
@@ -11,11 +50,85 @@ const DonationFormModal = () => {
     type: '',
     gender: '',
     petDescription: '',
-    imageUpload: ''
+    imageName: '',
+    imageUrl: ''
   })
+  const [createPet] = useMutation(CREATE_PET, {
+    refetchQueries: [
+      'getAllPets'
+    ]
+  });
+
+  if (!show) {
+      return null
+  }
+
+  const handlePic = (e) => {
+    setFormState({
+      ...formState,
+      imageName: e.target.value
+    });
+    const picture = e.target.files[0];
+    setFile(picture);
+  }
+
+  const uploadImage = () => {
```

Comment:
Wow. A lot of work went into this function but it works! Time well-spent. Lots of learning.